clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AbstractKafkaConsumerConfigProperties.java

```diff
@@ -0,0 +1,162 @@
+        this.valueDeserializerClass = valueDeserializerClass;
+    }
+
+    /**
+     * Sets the Kafka consumer config properties to be used.
+     *
+     * @param consumerConfig The config properties.
+     * @throws NullPointerException if the config is {@code null}.
+     */
+    public final void setConsumerConfig(final Map<String, String> consumerConfig) {
+        this.consumerConfig = Objects.requireNonNull(consumerConfig);
+    }
+
+    /**
+     * Checks if a configuration has been set.
+     *
+     * @return {@code true} if the {@value #PROPERTY_BOOTSTRAP_SERVERS} property has been configured with a non-null
+     *         value.
+     */
+    public final boolean isConfigured() {
+        return containsMinimalConfiguration(commonClientConfig) || containsMinimalConfiguration(consumerConfig);
+    }
+
+    /**
+     * Gets the Kafka consumer configuration. This includes changes made in {@link #adaptConfiguration(Map)}. The
+     * returned map will contain a property {@code client.id} which will be set to a unique value containing the already
+     * set client id or alternatively the value set via {@link #setDefaultClientIdPrefix(String)}, the given
+     * consumerName and a newly created UUID.
+     *
+     * Note: This method should be called for each new consumer, ensuring that a unique client id is used.
+     *
+     * @param consumerName A name for the consumer to include in the added {@code client.id} property.
+     * @return a copy of the consumer configuration with the applied properties or an empty map if neither a consumer
+     *         client configuration was set with {@link #setConsumerConfig(Map)} nor common configuration properties
+     *         were set with {@link #setCommonClientConfig(Map)}.
+     * @throws NullPointerException if consumerName is {@code null}.
+     */
+    public final Map<String, String> getConsumerConfig(final String consumerName) {
+        Objects.requireNonNull(consumerName);
+
+        if (commonClientConfig == null && consumerConfig == null) {
+            return new HashMap<>();
+        }
+
+        final Map<String, String> newConfig = new HashMap<>();
+        if (commonClientConfig != null) {
+            newConfig.putAll(commonClientConfig);
+        }
+        if (consumerConfig != null) {
+            newConfig.putAll(consumerConfig);
+        }
+
+        adaptConfiguration(newConfig);
+
+        Optional.ofNullable(keyDeserializerClass).ifPresent(serializerClass -> overrideConfigProperty(newConfig,
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, serializerClass.getName()));
+
+        Optional.ofNullable(valueDeserializerClass).ifPresent(serializerClass -> overrideConfigProperty(newConfig,
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, serializerClass.getName()));
+
+        setUniqueClientId(newConfig, consumerName, ConsumerConfig.CLIENT_ID_CONFIG);
+        return newConfig;
+    }
+
+    /**
+     * Adapt the properties.
+     * <p>
```

Comment:
would be good to know, when this method is being invoked so that subclasses can *do the right thing* :-)